tests/tests.cc

```diff
@@ -151,7 +171,166 @@ void connection_test::show_mav_state()
                   << local.y << ", " << local.z << std::endl;
     }
 
+    if (target_pos_global.is_initialized()) {
+        std::cout << "[connection test] Target Position Global: "
+                  << target_pos_global.lat << ", " << target_pos_global.lon
+                  << ", " << target_pos_global.alt << std::endl;
+
+        std::cout << "[connection test] Target Position: " << target_pos.x
+                  << ", " << target_pos.y << ", " << target_pos.z << std::endl;
+    }
+
     std::cout << "[connection test] Gps: " << (int)gps << std::endl;
 }
+
+conversion_test::conversion_test()
+{
+}
+
+conversion_test::~conversion_test()
+{
+}
+
+void conversion_test::run()
+{
+
+    // Execute test
+    while (true) {
+
+        // Set home position
+        mavlink_vehicles::global_pos_int home(-353631722, 1491651272, 574090);
+
+        // Set a new target position
+        mavlink_vehicles::local_pos target_local(10.0, 10.0, 10.0);
+        std::cout << "Local Position (before conversion): " << target_local.x
+                  << ", " << target_local.y << ", " << target_local.z
+                  << std::endl;
+
+        // Convert from local to global
+        mavlink_vehicles::global_pos_int target_global =
+            mavlink_vehicles::math::local_ned_to_global(target_local, home);
+        std::cout << "Global Position: " << target_global.lat << ", "
+                  << target_global.lon << ", " << target_global.alt
+                  << std::endl;
+
+        // Convert from global to local
+        target_local =
+            mavlink_vehicles::math::global_to_local_ned(target_global, home);
+        std::cout << "Local Position (after conversion): " << target_local.x
+                  << ", " << target_local.y << ", " << target_local.z
+                  << std::endl;
+
+        std::this_thread::sleep_for(
+            std::chrono::duration<int, std::milli>(2000));
+    }
+}
+
+
+mission_test::mission_test()
+{
+    // Socket Initialization
+    this->sock = socket(AF_INET, SOCK_DGRAM, IPPROTO_UDP);
+    if (sock == -1) {
+        perror("error opening socket");
+        exit(EXIT_FAILURE);
+    }
+
+    local_addr.sin_family = AF_INET;
+    local_addr.sin_addr.s_addr = htonl(INADDR_LOOPBACK);
+    local_addr.sin_port = htons(defaults::mavproxy_port);
+
+    if (bind(sock, (struct sockaddr *)&local_addr, sizeof(struct sockaddr)) ==
+        -1) {
+        perror("error bind failed");
+        close(sock);
+        exit(EXIT_FAILURE);
+    }
+
+    // Attempt to make it non blocking
+    if (fcntl(sock, F_SETFL, O_NONBLOCK) == -1) {
+        perror("error setting socket as nonblocking");
+        close(sock);
+        exit(EXIT_FAILURE);
+    }
+
+    // Instantiate mav_vehicle
+    this->mav = std::make_shared<mavlink_vehicles::mav_vehicle>(sock);
+}
+
+mission_test::~mission_test()
+{
+}
+
+void mission_test::run()
+{
+
+    // Initialize mavlink_vehicles update thread
+    this->send_recv_thread_run = true;
```

Comment:
It should turn false inside the mission_test destructor, in order to finish the thread before releasing the socket and the remaining resources. I need to fix it.